rascal-lsp/src/main/java/org/rascalmpl/vscode/lsp/uri/FallbackResolver.java

```diff
@@ -250,5 +272,45 @@ public void unwatch(ISourceLocation root, Consumer<ISourceLocationChanged> watch
         getClient().removeWatcher(root, watcher, getServer());
 
     }
+    
+    public boolean isFileManaged(ISourceLocation file) {
+        for (final var service : textDocumentServices) {
+            if (service.isManagingFile(file)) {
+                return true;
+            }
+        }
+        return false;
+    }
 
+    @Override
+    public ISourceLocation resolve(ISourceLocation input) throws IOException {
+        if (isFileManaged(input)) {
+            try {
+                return URIUtil.changeScheme(input, "lsp+" + input.getScheme()).top();
```

Comment:
why the `.top`? that would loose information on the URI that we cannot recover?

Reply:
The offset/length information is restored by `resolveAndFixOffsets`. Not clearing it here would result in "double offset". I've added a comment